state-chain/runtime/src/chainflip.rs

```diff
@@ -1024,3 +1024,31 @@ impl cf_traits::MinimumDeposit for MinimumDepositProvider {
 		}
 	}
 }
+
+pub struct LpOrderCallIndexer;
+impl CallIndexer<RuntimeCall> for LpOrderCallIndexer {
```

Comment:
I'm not sure this way of abstracting it improves things. We always have to pass in the indexer to the flip pallet. If we were to index other things for fee scaling, would we put in a separate CallIndex instance, with a separate associated type in on the pallet?

I think it makes more sense to have the more globally named and instantiated call indexer, that's for all calls, rather than an instance for each.

Reply:
For now it only applies to LP orders, and I wanted to avoid naming everything `CallIndexer`. The intention wasn't necessarily that we would have multiple of these.

Reply:
Fair